src/common/tools/dev_server.ts

```diff
@@ -150,14 +150,17 @@ app.get('/out/**/*.js', async (req, res, next) => {
   const tsUrl = jsUrl.replace(/\.js$/, '.ts');
   if (compileCache.has(tsUrl)) {
     res.setHeader('Content-Type', 'application/javascript');
+    res.setHeader('Service-Worker-Allowed', '/');
     res.send(compileCache.get(tsUrl));
     return;
   }
 
-  let absPath = path.join(srcDir, tsUrl);
+  // I'm not sure if this is the way I should handle it...
```

Comment:
This would require `out` to be built in order to use the dev server. Dev server is supposed to work without any build step. So instead when we get a `.worker.js` request we want to generate the response on the fly similar to what is done for `listing.js` (except we don't need to cache it because the generation should be very fast).

Reply:
I wish I'd understand that bit ;)
Can you expand on that?

Reply:
Yes, sorry. For the on-disk builds, we generate a bunch of files and write them to disk. However, the dev server does all building in-memory. When the webpage requests a .js file, we builds the .ts file. When it requests a listing.js file, we generate one by crawling the src directory for tests. Similarly here when it requests a .worker.js file, we need to generate one in the same way 

The generation code of course should be shared between the dev server and the gen_listings_and_webworkers script.

Reply:
IIUC (which is not likely) gen_listings_and_webworkers does not generate code. It writes to a file that is then handled by dev_server and processed. How would the generation code work there?

Reply:
`gen_listings_and_webworkers` is only used in the offline build, it shouldn't affect the dev server at all. It generates `gen/*/listing.js` and `gen/*/webworker/**/*.worker.js`. Those are then copied into the out directories by Grunt.

In the dev server instead we have a special request handler when a `listing.js` file is requested:
`app.get('/out/:suite([a-zA-Z0-9_-]+)/listing.js', async (req, res, next) => {`
and we generate the listing and return it, instead of using the on-disk `listing.js`.

Reply:
Thank you for bearing with me @kainino0x. I think I now understand what you were telling me (for a long time). Let me know if latest commit looks good to you.

Reply:
No problem, looks good!